src/main/java/org/sagebionetworks/web/client/presenter/LoginPresenter.java

```diff
@@ -46,79 +46,22 @@ public class LoginPresenter
   private AuthenticationController authenticationController;
   private GlobalApplicationState globalApplicationState;
   private SynapseAlert synAlert;
-  private PopupUtilsView popupUtils;
 
   @Inject
   public LoginPresenter(
     LoginView view,
     AuthenticationController authenticationController,
     GlobalApplicationState globalApplicationState,
-    SynapseAlert synAlert,
-    PopupUtilsView popupUtils
+    SynapseAlert synAlert
   ) {
     this.view = view;
     this.authenticationController = authenticationController;
     this.globalApplicationState = globalApplicationState;
     this.synAlert = synAlert;
-    this.popupUtils = popupUtils;
     view.setSynAlert(synAlert);
     view.setPresenter(this);
   }
 
-  @Override
-  public void onAcceptTermsOfUse() {
```

Comment:
So if I sign in to OneSage and have not accepted the terms of use, that app already handles prompting today (before new ToS implementation)?

Reply:
It should.  The ApplicationSessionManager attempts to initialize the session (used in OneSage AppWrapper).  In doing so, it tries to get your User Profile, and looks for the "Terms of use have not been signed" error.  It sets the ApplicationSessionContext acceptsTermsOfUse value.  If this value is false, then in AppInitializer, it will redirect you to /authenticated/signTermsOfUse, showing you the TermsOfUsePage